File: Semestres/11.Febrero-Junio 2022/EstructurasDatosUSB/corte2/Tema21RecorridosArboles/Tree3.java
import java.util.LinkedList;
import java.util.Arrays;

public class Tree3 {
    private String key;
    private Tree3 left;
    private Tree3 right1;
    private Tree3 right2;
    public Tree3(String key, Tree3 left, Tree3 right1, Tree3 right2) {
        this.key = key;
        this.left = left;
        this.right1 = right1;
        this.right2 = right2;
    }
    public String getKey() {
        return key;
    }
    public void setKey(String key) {
        this.key = key;
    }
    public Tree3 getLeft() {
        return left;
    }
    public void setLeft(Tree3 left) {
        this.left = left;
    }
    public Tree3 getRight1() {
        return right1;
    }
    public void setRight1(Tree3 right1) {
        this.right1 = right1;
    }
    public Tree3 getRight2() {
        return right2;
    }
    public void setRight2(Tree3 right2) {
        this.right2 = right2;
    }

    public LinkedList<String> preorder(){
        LinkedList<String> list = new LinkedList<String>();
        list.add(this.key);
        if(this.left!=null) list.addAll(this.left.preorder());
        if(this.right1!=null)list.addAll(this.right1.preorder());
        if(this.right2!=null)list.addAll(this.right2.preorder());
        return list;
    }

    public LinkedList<String> inorder(){
        LinkedList<String> list = new LinkedList<String>();
        if(this.left!=null) list.addAll(this.left.inorder());
        list.add(this.key);
        if(this.right1!=null)list.addAll(this.right1.inorder());
        if(this.right2!=null)list.addAll(this.right2.inorder());
        return list;
    }

    public LinkedList<String> posorder(){
        LinkedList<String> list = new LinkedList<String>();

        if(this.left!=null) list.addAll(this.left.posorder());
        if(this.right1!=null)list.addAll(this.right1.posorder());
        if(this.right2!=null)list.addAll(this.right2.posorder());
        list.add(this.key);
        return list;
    }
}

File: Semestres/11.Febrero-Junio 2022/EstructurasDatosUSB/corte2/Tema21RecorridosArboles/Tree3.py
class Tree:
    def __init__(self, key, left, right1, right2):
        self.key = key
        self.left = left
        self.right1 = right1
        self.right2 = right2

    def preorder(self):
        salida, leftL, right1L, right2L = [], [], [], []
        salida.append(self.key)
        if(self.left != None):
            leftL = self.left.preorder()
        if(self.right1 != None):
            right1L = self.right1.preorder()
        if(self.right2 != None):
            right2L = self.right2.preorder()

        for l in leftL:
            salida.append(l)

        for r1 in right1L:
            salida.append(r1)

        for r2 in right2L:
            salida.append(r2)

        return salida

    def inororder(self):
        salida, leftL, right1L, right2L = [], [], [], []

        if(self.left != None):
            leftL = self.left.inororder()

        for l in leftL:
            salida.append(l)

        salida.append(self.key)

        if(self.right1 != None):
            right1L = self.right1.inororder()
        if(self.right2 != None):
            right2L = self.right2.inororder()

        for r1 in right1L:
            salida.append(r1)

        for r2 in right2L:
            salida.append(r2)

        return salida

    def posorder(self):
        salida, leftL, right1L, right2L = [], [], [], []

        if(self.left != None):
            leftL = self.left.posorder()

        for l in leftL:
            salida.append(l)

        
        if(self.right1 != None):
            right1L = self.right1.posorder()
        if(self.right2 != None):
            right2L = self.right2.posorder()

        for r1 in right1L:
            salida.append(r1)

        for r2 in right2L:
            salida.append(r2)

        salida.append(self.key)

        return salida
